app/frontend/wallet/blockchain-explorer.ts

```diff
@@ -170,6 +179,114 @@ const blockchainExplorer = (ADALITE_CONFIG) => {
     })
   }
 
+  async function getPoolInfo(url) {
+    const response = await request(
+      `${ADALITE_CONFIG.ADALITE_SERVER_URL}/api/poolMeta`,
+      'POST',
+      JSON.stringify({poolUrl: url}),
+      {'Content-Type': 'application/json'}
+    ).catch(() => {
+      return {}
+    })
+    return response
+  }
+
+  async function getDelegationHistory(accountPubkeyHex, validStakepools) {
+    // Urls
+    const delegationsUrl = `https://explorer-staging.adalite.io/api/account/delegationHistory/${accountPubkeyHex}`
+    // const rewardsUrl = `https://explorer-staging.adalite.io/api/account/rewardHistory/${accountPubkeyHex}`
+    const withdrawalsUrl = `https://explorer-staging.adalite.io/api/account/withdrawalHistory/${accountPubkeyHex}`
+
+    // Run requests in parallel
+    const delegationsRequest = request(delegationsUrl).catch(() => [])
+    // const rewardsRequest = request(rewardsUrl).catch(()=>[]
+    const withdrawalsRequest = request(withdrawalsUrl).catch(() => [])
+
+    // Await all
+    const delegations = await delegationsRequest
+    const rewards = [] //await rewardsRequest
+    const withdrawals = await withdrawalsRequest
+
+    const extractUrl = (poolHash) =>
+      validStakepools[poolHash] ? validStakepools[poolHash].url : null
+
+    const poolMetaUrls = distinct(
+      [...delegations, ...rewards].map(({poolHash}) => extractUrl(poolHash))
+    ).filter((x) => x != null)
+
+    const metaUrlToPoolNameMap = (await Promise.all(
+      poolMetaUrls.map((url: string) =>
+        getPoolInfo(url).then((metaData) => ({url, name: metaData.name}))
+      )
+    )).reduce((map, {url, name}) => {
+      map[url] = name
+      return map
+    }, {})
+
+    const poolHashToPoolName = (poolHash) => {
+      const poolName = metaUrlToPoolNameMap[extractUrl(poolHash)]
+      return poolName || UNKNOWN_POOL_NAME
+    }
+
+    const parseStakePool = (delegationHistoryObject) => {
+      const stakePool: StakePool = {
+        id: delegationHistoryObject.poolHash,
+        name: poolHashToPoolName(delegationHistoryObject.poolHash),
+      }
+
+      return stakePool
+    }
```

Comment:
This can be shortened to something like 
```
const parseStakePool = (delegationHistoryObject) => ({
  id: delegationHistoryObject.poolHash,
  name: poolHashToPoolName(delegationHistoryObject.poolHash),
}: StakePool)
```

Reply:
Refactored